### PR TITLE
feat: initial module

### PR DESCRIPTION
**Pull request by MatthewLemmond**
### Description
Initial version of the module to create db2 instances

### Release required?

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

Initial working release of the module for testing the DB2 resource instance creation

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: .catalog-onboard-pipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+apiVersion: v1
+offerings:
+  - name: deploy-arch-ibm-mq-cloud
+    kind: solution
+    catalog_id: 7df1e4ca-d54c-4fd0-82ce-3d13247308cd
+    offering_id: 30a698d4-ab56-4668-bc01-03cd2135d7e4 MQ?
```
Comment (ocofaigh) [inline .catalog-onboard-pipeline.yaml:7]: This is `90a14f92-4b86-4b27-acc1-dd6eefe0bfe9`

File: .catalog-onboard-pipeline.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+apiVersion: v1
+offerings:
+  - name: deploy-arch-ibm-mq-cloud
```
Comment (shemau) [inline .catalog-onboard-pipeline.yaml:4]: This will be my copy and paste error

**Review by shemau — CHANGES_REQUESTED** [review]
Do we want both the basic and advanced example? The basic one does not have a readme.md if it is being kept.

File: version.tf [hunk 3]
```diff
@@ -3,10 +3,10 @@ terraform {
...
-  #  required_providers {
-  #    ibm = {
-  #      source  = "IBM-Cloud/ibm"
-  #      version = ">= 1.64.0, < 2.0.0"
-  #    }
-  #  }
+  required_providers {
+    ibm = {
+      source  = "IBM-Cloud/ibm"
+      version = ">= 1.72.1"
```
Comment (ocofaigh) [inline version.tf:9]: why did you remove the upper limit? Please add it back `">= 1.72.1, < 2.0.0"`

File: README.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <!-- Update this title with a descriptive name. Use sentence case. -->
-# Terraform modules template project
+# DB2 project
```
Comment (ocofaigh) [inline README.md:2]: can we have a better title?

**Review by ocofaigh — CHANGES_REQUESTED** [review]
The tests need to test the DA (standard and upgrade). Thats a big gap. Especially since the provider visibility is set to private. Your going to need to test this in schematics

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**